app/src/main/java/com/github/se/polyfit/ui/screen/MapScreen.kt

```diff
@@ -44,68 +51,83 @@ val MAX_SIZE_RAYON = 2000f
+
   val currentLocation = remember { mutableStateOf(Location.default()) }
-  val mutex = Mutex()
+
+  var selectedPost by remember { mutableStateOf<Post?>(null) }
 
   var isCircleShown by remember { mutableStateOf(false) }
 
+  var mainLaunchedIsFinished by remember { mutableStateOf(false) }
+
   val cameraPositionState = rememberCameraPositionState {
     position = CameraPosition.fromLatLngZoom(LatLng(EPFL_LATITUDE, EPFL_LONGITUDE), 15f)
   }
+
   var sliderPosition by remember { mutableFloatStateOf((MIN_SIZE_RAYON + MAX_SIZE_RAYON) / 2) }
 
-  LaunchedEffect(currentLocation) {
-    mutex.withLock {
-      currentLocation.value = mapviewMap.getCurrentLocation()
-      mapviewMap.setLocation(currentLocation.value)
-    }
+  /**
+   * Effect to be launched when the mapViewModel changes. Initializes the current location and sets
+   * up the map and post listeners.
+   *
+   * This effect fetches the current location asynchronously, sets the location in the mapViewModel,
+   * and begins listening to posts. It also updates the camera position of the map and observes
+   * posts to update the state.
+   *
+   * @param mapViewModel The ViewModel that handles map-related data and operations.
+   */
+  LaunchedEffect(key1 = mapViewModel) {
+    currentLocation.value = mapViewModel.getCurrentLocation().await()
+    mapViewModel.setLocation(currentLocation.value)
+
+    mapViewModel.listenToPosts()
+    mapViewModel.posts.observeForever { posts.value = it }
 
-    mapviewMap.listenToPosts()
-    mapviewMap.posts.observeForever { posts.value = it }
-  }
-  LaunchedEffect(sliderPosition) {
-    isCircleShown = true
-    mutex.withLock { mapviewMap.setRadius(sliderPosition.toDouble() / 1000) }
+    cameraPositionState.position =
+        CameraPosition.fromLatLngZoom(
+            LatLng(currentLocation.value.latitude, currentLocation.value.longitude), 15f)
 
-    if (isCircleShown) {
-      mapviewMap.listenToPosts()
-      mapviewMap.posts.observeForever { posts.value = it }
-    }
-    delay(3000)
-    isCircleShown = false
+    mainLaunchedIsFinished = true
   }
 
-  LaunchedEffect(key1 = mapviewMap) {
-    //        mapviewMap.getAllPost().collect { posts.value = it }
-
-    Log.i("Map", "MapScreen $posts")
-    mutex.withLock {
-      currentLocation.value = mapviewMap.getCurrentLocation()
-      mapviewMap.setLocation(currentLocation.value)
+  /**
+   * Handles changes in the slider position. Updates the slider position and adjusts the map's
+   * search radius accordingly. If the main process has finished, it sets the new radius and
+   * triggers an update to listen for posts.
+   *
+   * @param scope The CoroutineScope in which to launch the coroutine.
+   * @param newPosition The new position of the slider.
+   */
+  val mutex = Mutex()
+  fun onSliderChanged(scope: CoroutineScope) {
+    scope.launch {
+      mutex.withLock {
+        if (mainLaunchedIsFinished) {
+
+          mapViewModel.setRadius(sliderPosition.toDouble() / 1000)
```

Comment:
It is useful to have the units
```suggestion
          mapViewModel.setRadius(sliderPositionMeter.toDouble() / 1000)
```